spec/system/admin/barcode_items_system_spec.rb

```diff
@@ -1,40 +1,95 @@
-RSpec.describe "Barcode Items Admin", type: :system, js: true do
-  before do
-    sign_in(@super_admin)
-  end
-
+RSpec.describe "Barcode Items Admin", type: :system do
   context "while signed in as a super admin" do
     before do
-      visit admin_barcode_items_path
+      sign_in(@super_admin)
     end
 
-    let(:item) { create(:item) }
-    let!(:barcode_item) { create(:global_barcode_item) }
-    it "should create a new global barcode" do
-      barcode_value = 66_666
-      click_on "Add New Barcode"
-      fill_in "Quantity", with: 100
-      select item.base_item.name, from: "barcode_item_barcodeable_id"
-      fill_in "Barcode", with: barcode_value
-      click_on "Save"
-      expect(page).to have_content(barcode_value)
-      expect(page).to have_content("Tampons")
-      expect(page).to have_content("100")
+    context "user visits the index page" do
+      before do
+        visit admin_barcode_items_path
+      end
+
+      let!(:item) { create(:item) }
+      let!(:base_item) { create(:base_item) }
+      let!(:barcode_item) { create(:global_barcode_item) }
+
+      it 'shows the barcode item' do
+        page.refresh # not sure what is making it need this
```

Comment:
@perezperret I figured out why in this case you needed the `page.refresh` to get the specs to work (it as a bit tricky!). The problem was that `visit admin_barcode_items_path` was being executed before the `let!`s. This is why the page when it was being rendered didn't have the things you've defined in `let!`.

A solution would be to move the `let!` above the `before/do` block that way `let!` is being called before the visit to `admin_barcode_items_path`

Reply:
Awesome! Thanks for your help! Should be good now.